src/Stream.php

```diff
@@ -208,7 +208,17 @@ public function read($length)
             throw new \RuntimeException('Cannot read from non-readable stream');
         }
 
-        return fread($this->stream, $length);
+        if (0 === $length) {
+            return '';
+        }
+
+        $string = $length > 0 ? fread($this->stream, $length) : false;
```

Comment:
Please throw an explicit RuntimeException when $length < 0 saying that "Length parameter cannot be negative".